subprojects/core/src/main/java/org/gradle/internal/fingerprint/GenericFileTreeSnapshotter.java

```diff
@@ -16,9 +16,15 @@
 
 package org.gradle.internal.fingerprint;
 
+import org.gradle.api.file.FileVisitor;
 import org.gradle.api.internal.file.FileTreeInternal;
 import org.gradle.internal.snapshot.FileSystemSnapshot;
 
+/**
+ * A snapshotter for generic file trees, which are not based on a directory on disk.
+ *
+ * This is needed to build a Merkle directory tree from the elements of a file tree obtained by {@link org.gradle.api.file.FileTree#visit(FileVisitor)}.
+ */
```

Comment:
I think it would be important to give examples of what these trees might be.

Reply:
I added an example. Finally, I can only come up with the `TarFileTree` one...